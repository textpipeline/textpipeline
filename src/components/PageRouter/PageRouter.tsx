import NotFound from 'components/NotFound';
import React from 'react';
import { Route, Switch } from 'react-router-dom';
import routes from 'routes';

const PageRouter: React.FC = () => (
  <Switch>
    {Object.keys(routes).map(routeName => (
      <Route key={routeName} {...routes[routeName]} />
    ))}
    <Route path="*">
      <NotFound />
    </Route>
  </Switch>
);

export default PageRouter;
