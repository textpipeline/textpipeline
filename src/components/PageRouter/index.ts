export { default } from './PageRouter';
export * from './PageRouter';
