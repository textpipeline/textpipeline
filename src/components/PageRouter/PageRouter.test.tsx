import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import PageRouter from './PageRouter';

describe('PageRouter', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<PageRouter />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
