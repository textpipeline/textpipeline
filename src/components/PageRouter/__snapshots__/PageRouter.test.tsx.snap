// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageRouter renders correctly 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    key="home"
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    key="oss"
    path="/oss"
  />
  <Route
    component={[Function]}
    exact={true}
    key="transforms"
    path="/transforms"
  />
  <Route
    component={[Function]}
    exact={true}
    key="Binary To Decimal"
    path="/transforms/binary-to-decimal"
  />
  <Route
    component={[Function]}
    exact={true}
    key="decimal.js-light"
    path="/oss/decimal-js-light"
  />
  <Route
    path="*"
  >
    <NotFound />
  </Route>
</Switch>
`;
